defaults/main.yml

```diff
@@ -67,11 +67,12 @@ dataverse:
     logformat: ulf
     adminuser: admin
     adminpass: notPr0d
+    siteurl:
     timeout: 180
     root: /usr/local
     dir: glassfish4 # change this for payara, etc.
```

Comment:
Is it safe to set `dir` to `payara5`? I'm trying it. :smile:

Reply:
I moved siteurl to make the two defaults files match. I don't see glassfish4 hard-coded anywhere, so... should be safe?

Reply:
I just tried payara5 as the dir and the zip url like this:

```
<     dir: glassfish4 # change this for payara, etc.
<     zipurl: http://dlc-cdn.sun.com/glassfish/4.1/release/glassfish-4.1.zip
<     #zipurl: https://search.maven.org/remotecontent?filepath=fish/payara/distributions/payara/5.192/payara-5.192.zip
---
>     dir: payara5
>     zipurl: https://search.maven.org/remotecontent?filepath=fish/payara/distributions/payara/5.192/payara-5.192.zip
```

... but I got this error:


fatal: [localhost]: FAILED! => {"changed": true, "cmd": "nohup /usr/local/payara5/bin/asadmin deploy /tmp/dvinstall/dataverse.war", "delta": "0:00:45.493531", "end": "2019-06-24 13:40:06.384827", "msg": "non-zero return code", "rc": 1, "start": "2019-06-24 13:39:20.891296", "stderr": "remote failure: Error occurred during deployment: Exception while loading the app : javax.ejb.CreateException: Initialization failed for Singleton StartupFlywayMigrator. Please see server.log for more details.", "stderr_lines": ["remote failure: Error occurred during deployment: Exception while loading the app : javax.ejb.CreateException: Initialization failed for Singleton StartupFlywayMigrator. Please see server.log for more details."], "stdout": "Command deploy failed.", "stdout_lines": ["Command deploy failed."]}

server.log says this:

Caused by: org.flywaydb.core.internal.sqlscript.FlywaySqlScriptException: 
Migration V4.14.0.2__2043-split-gbr-table.sql failed
----------------------------------------------------
SQL State  : 42703
Error Code : 0
Message    : ERROR: column "downloadtype" does not exist
  Hint: There is a column named "downloadtype" in table "filedownload", but it cannot be referenced from this part of the query.
  Position: 100
Location   : db/migration/V4.14.0.2__2043-split-gbr-table.sql (/usr/local/payara5/glassfish/domains/domain1/applications/dataverse/WEB-INF/classes/db/
migration/V4.14.0.2__2043-split-gbr-table.sql)
Line       : 2
Statement  : insert into filedownload(GUESTBOOKRESPONSE_ID,DOWNLOADTYPE,DOWNLOADTIMESTAMP,SESSIONID) select ID, DOWNLOADTYPE,RESPONSETIME,SESSIONID fr
om guestbookresponse

This probably has more to do with https://github.com/IQSS/dataverse/pull/5863 than anything in this ansible repo.

@donsizemore can you please remind me how to have dataverse-ansible install a release rather than whatever is in develop? I feel like lately I always get develop no matter what I try. We can talk this out in IRC if it's easier.

Reply:
@pdurbin yeah, I get the same deployment failure under payara5.

You just set dataverse_branch to "release" (which I thought was the default?) and the role will use the dvinstall.zip specified by the version variable.

Reply:
>  Migration V4.14.0.2__2043-split-gbr-table.sql failed

I just opened this issue: https://github.com/IQSS/dataverse/issues/5964 (guestbookresponse table: "ERROR: column "downloadtype" does not exist")